code/game/objects/machinery/telecomms/broadcasting.dm

```diff
@@ -199,6 +199,48 @@
 			continue
 		hearer.Hear(rendered, virt, language, message, frequency, spans)
 
+
+	//Check that speaker isn't banned from TTS
+	var/atom/movable/speaker = virt?.source
+	var/banned = FALSE
+	if(ismob(speaker))
+		var/mob/potential_user = speaker
+		if(is_banned_from(potential_user.ckey, "TTS") || potential_user.client?.prefs.muted & MUTE_TTS)
+			banned = TRUE
+
+	//If they aren't, handle radio TTS
+	if(speaker && speaker.voice && !banned)
+		var/is_speaker_squad_lead = FALSE
+		if(ishuman(speaker))
+			var/mob/living/carbon/human/human_speaker = speaker
+			if(human_speaker.assigned_squad?.squad_leader == speaker)
+				is_speaker_squad_lead = TRUE
+
+		var/is_speaker_command_freq = FALSE
+		if(frequency == FREQ_COMMAND || frequency == FREQ_COMMAND_SOM)
+			is_speaker_command_freq = TRUE
+
+		if(speaker in receive)
+			receive -= speaker //This list isn't used again, so we can just cut out the original speaker from it so TTS doesn't play twice
+
+		for(var/mob/living/carbon/human/potential_hearer in receive)
+			if(potential_hearer.stat >= UNCONSCIOUS || potential_hearer.disabilities & DEAF || !(potential_hearer.client?.prefs.sound_tts != TTS_SOUND_OFF))
+				continue
+
+			var/radio_flags = potential_hearer.client.prefs.radio_tts_flags
+			var/should_play_tts = FALSE
+			if(CHECK_BITFIELD(radio_flags, RADIO_TTS_ALL))
+				should_play_tts = TRUE
+			else if(potential_hearer.assigned_squad?.radio_freq == frequency && (CHECK_BITFIELD(radio_flags, RADIO_TTS_SQUAD) || (CHECK_BITFIELD(radio_flags, RADIO_TTS_SL) && is_speaker_squad_lead)))
+				should_play_tts = TRUE
+			else if(CHECK_BITFIELD(radio_flags, RADIO_TTS_COMMAND) && is_speaker_command_freq)
+				should_play_tts = TRUE
+
+			if(!should_play_tts)
+				continue
+
+			INVOKE_ASYNC(SStts, TYPE_PROC_REF(/datum/controller/subsystem/tts, queue_tts_message), potential_hearer, html_decode(message), language, speaker.voice, potential_hearer.voice_filter, local = TRUE, pitch = speaker.pitch, special_filters = TTS_FILTER_RADIO)
```

Comment:
You'll instead want to modify the backend so that you can pass in an option to disable directionality for a specific request. This is generally a lot more inefficient than just passing in a list of potential hearers as you're making a lot of HTTP requests when you only really need to make 1. Also, it'll fall into the same problems of the same person speaking where they're able to speak over themselves, which can kinda hurt the illusion of TTS.

Best thing to do is to just add another option to queue_tts_message that lets you set some `directionality` var to FALSE. Pass that into the TTS request, then in the subsystem before `play_tts`, pass that in and disable directionality/range checks if it is set to FALSE.

Reply:
Alternatively, you could just add them all up into a list and send it to TTS subsystem in one go.
Add a variable for `directionality`
If it's set, ignore turf source and play at source.